services/storage/src/simcore_service_storage/api/rest/_files.py

```diff
@@ -0,0 +1,371 @@
+    user_agent: Annotated[str | None, Header()],
+    request: Request,
+):
+    dsm = get_dsm_provider(request.app).get(location_id)
+    try:
+        data = await dsm.get_file(
+            user_id=query_params.user_id,
+            file_id=file_id,
+        )
+    except FileMetaDataNotFoundError:
+        # NOTE: LEGACY compatibility
+        # This is what happens Larry... data must be an empty {} or else some old dynamic services will FAIL (sic)
+        # Cannot remove until we retire all legacy services
+        # https://github.com/ITISFoundation/osparc-simcore/issues/5676
+        # https://github.com/ITISFoundation/osparc-simcore/blob/cfdf4f86d844ebb362f4f39e9c6571d561b72897/services/storage/client-sdk/python/simcore_service_storage_sdk/models/file_meta_data_enveloped.py#L34
+        return Envelope[dict](
+            data={},
+            error="No result found",  # NOTE: LEGACY compatibility
+        )
+
+    if user_agent == "OpenAPI-Generator/0.1.0/python":
+        # NOTE: LEGACY compatiblity with API v0.1.0
+        # SEE models used in sdk in:
+        # https://github.com/ITISFoundation/osparc-simcore/blob/cfdf4f86d844ebb362f4f39e9c6571d561b72897/services/storage/client-sdk/python/simcore_service_storage_sdk/models/file_meta_data_enveloped.py#L34
+        # https://github.com/ITISFoundation/osparc-simcore/blob/cfdf4f86d844ebb362f4f39e9c6571d561b72897/services/storage/client-sdk/python/simcore_service_storage_sdk/models/file_meta_data_type.py#L34
+        return Envelope[FileMetaDataGetv010](
+            data=FileMetaDataGetv010(
+                file_uuid=data.file_uuid,
+                location_id=data.location_id,
+                location=data.location,
+                bucket_name=data.bucket_name,
+                object_name=data.object_name,
+                project_id=data.project_id,
+                project_name=data.project_name,
+                node_id=data.node_id,
+                node_name=data.node_name,
+                file_name=data.file_name,
+                user_id=data.user_id,
+                user_name=None,
+            )
+        )
+
+    return Envelope[FileMetaDataGet](data=FileMetaDataGet(**data.model_dump()))
+
+
+@router.get(
+    "/locations/{location_id}/files/{file_id:path}",
+    response_model=Envelope[FileDownloadResponse],
+)
+async def download_file(
+    location_id: LocationID,
+    file_id: StorageFileID,
+    query_params: Annotated[FileDownloadQueryParams, Depends()],
+    request: Request,
+) -> Envelope[FileDownloadResponse]:
+    dsm = get_dsm_provider(request.app).get(location_id)
+    link = await dsm.create_file_download_link(
+        query_params.user_id, file_id, query_params.link_type
+    )
+    return Envelope[FileDownloadResponse](data=FileDownloadResponse(link=link))
+
+
+@router.put(
+    "/locations/{location_id}/files/{file_id:path}",
+    response_model=Envelope[FileUploadResponseV1] | Envelope[FileUploadSchema],
+)
+async def upload_file(
```

Comment:
maybe this should also have cancel on disconnect decorator?

Reply:
good point. will add a middleware actually.